src/components/Sidebars/Payback.js

```diff
@@ -11,6 +19,45 @@ import Info from './shared/Info';
 import InfoContainer from './shared/InfoContainer';
 import lang from 'languages';
 import { MAX_UINT_BN } from '../../utils/units';
+import Loader from '../Loader';
+
+const initialState = {
+  loaderToggle: false,
+  allowanceSet: false,
+  userHasAllowance: null,
+  initialised: false
+};
+
+const reducer = (state, action) => {
+  console.log(action);
+  const { type, payload } = action;
+  switch (type) {
+    case 'loader-on':
+      return {
+        ...state,
+        loaderToggle: true
+      };
+    case 'loader-off':
+      return {
+        ...state,
+        loaderToggle: false
+      };
+    case 'initial-allowance':
+      return {
+        ...state,
+        initialised: true,
+        userHasAllowance: payload.allowanceSet,
```

Comment:
the distinction b/w `userHasAllowance` and `allowanceSet` is a little fuzzy to me. is there anything you think we could do to sharpen that?

Reply:
I agree that it does need some clarification. `userHasAllowance` determines whether or not to render the toggle in the first place. Should the user have the dai allowance set, then the toggle should not render. If the dai allowance is not set, the toggle should render as inactive, when clicked and the transaction is processed, it should change to active. However, during that session, the toggle should persist as active which I think is the most intuitive for the user.

Would it be more the naming of these variables over the implementation? I suppose `userHasAllowance -> userHasInitialAllowance` may be an improvement but I feel it's a bit long. I also wish js enabled the use of `'?'` characters to indicate boolean variables better, `allowanceSet < allowanceSet?`